pkg/route/here_planner.go

```
@@ -133,38 +141,53 @@ type RoutingResponseBody struct {
-			return 0, errors.Wrap(err, "bad Here response, bad body read")
+		} else {
+			p.logger.Info("Got non-200 response from HERE routing.", zap.Int("http_status", resp.StatusCode), zap.String("here_error", string(bodyBytes)))
 		}
-		p.logger.Info("Got non-200 response from HERE routing.", zap.Int("http_status", resp.StatusCode), zap.String("here_error", string(bodyBytes)))
-		return 0, errors.New("error response from HERE")
+
+		if resp.StatusCode >= 400 && resp.StatusCode < 500 {
+			// Client error
+			return 0, NewUnroutableRouteError(resp.StatusCode, source, dest)
+		}
+		return 0, NewUnknownRoutingError(resp.StatusCode, source, dest)
 	} else {
-		routeDecoder := json.NewDecoder(resp.Body)
-		var response RoutingResponseBody
-		err = routeDecoder.Decode(&response)
+		distanceMiles, err := getDistanceMiles(resp.Body)
 		if err != nil {
-			p.logger.Error("Failed to decode response from HERE routing.", zap.Error(err))
-			return 0, errors.Wrap(err, "decoding routing response from HERE")
-		} else if len(response.Response.Routes) == 0 {
-			p.logger.Error("Expected at least one route in HERE routing response", zap.Error(err))
-			return 0, errors.New("no Route in HERE routing response")
-		} else {
-			return int(math.Round(float64(response.Response.Routes[0].Summary.Distance) / metersInAMile)), nil
+			p.logger.Error("Failed to decode response from HERE routing.", zap.Error(err), zap.Any("source", source), zap.Any("destination", dest))
+
 		}
+		return distanceMiles, err
 	}
 }
 
 func (p *herePlanner) Zip5TransitDistance(source string, destination string) (int, error) {
-	return zip5TransitDistanceHelper(p, source, destination)
+	distance, err := zip5TransitDistanceHelper(p, source, destination)
+	if err != nil {
+		p.logger.Error("Failed to calculate HERE route between ZIPs", zap.String("source", source), zap.String("destination", destination))
+	}
+
+	return distance, nil
```

I assume you meant to return `err` instead of `nil` here -- we'll swallow the error otherwise.